neuro_san/registries_watcher/periodic_updater/manifest_periodic_updater.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+import logging
+import time
+import threading
+from typing import Dict
+
+from neuro_san.internals.tool_factories.service_tool_factory_provider import ServiceToolFactoryProvider
+from neuro_san.registries_watcher.periodic_updater.registry_observer import RegistryObserver
+from neuro_san.internals.graph.registry.agent_tool_registry import AgentToolRegistry
+from neuro_san.internals.graph.persistence.registry_manifest_restorer import RegistryManifestRestorer
+
+
+class ManifestPeriodicUpdater:
+    """
+    Class implementing periodic manifest directory updates
+    by watching agent files and manifest file itself.
+    """
+
+    def __init__(self, manifest_path: str, update_period: int):
+        """
+        Constructor.
+        :param manifest_path: file path to server manifest file
+        :param update_period: update period in seconds
+        """
+        self.manifest_path: str = manifest_path
+        self.update_period: int = update_period
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.updater = threading.Thread(target=self._run, daemon=True)
```

Comment:
"updater" is our internal thread, that will wake periodically and check if anything has happened
in the registry directory we are watching.